src/ess.jl

```diff
@@ -266,10 +268,17 @@ function ess_rhat(
     # when chains have mixed poorly anyways.
     # leave the last even autocorrelation as a bias term that reduces variance for
     # case of antithetical chains, see below
-    maxlag = min(maxlag, niter - 4)
-    if !(maxlag > 0) || T === Missing
-        return similar(chains, Missing, axes_out), similar(chains, Missing, axes_out)
+    if !(niter > 4)
+        throw(ArgumentError("number of draws after splitting must >4 but is $niter."))
     end
+    maxlag > 0 || throw(DomainError(maxlag, "maxlag must be >0."))
+    maxlag = min(maxlag, niter - 4)
```

Comment:
Maybe move the checks to the top of the function?

Reply:
I like keeping these checks together and using `niter`, which is computed with the other sizes, so I prefer to leave here.